ui/apps/platform/src/services/ComplianceScanConfigurationService.ts

```diff
@@ -0,0 +1,165 @@
+import axios from 'services/instance';
+import qs from 'qs';
+
+import { ApiSortOption } from 'types/search';
+import { SlimUser } from 'types/user.proto';
+
+import { complianceV2Url } from './ComplianceCommon';
+import { CancellableRequest, makeCancellableAxiosRequest } from './cancellationUtils';
+import { Empty } from './types';
+
+const complianceScanConfigBaseUrl = `${complianceV2Url}/scan/configurations`;
+
+export type ScheduleBase = {
+    hour: number;
+    minute: number;
+};
+
+export type UnsetSchedule = ScheduleBase & {
+    intervalType: 'UNSET';
+};
+
+export type DailySchedule = ScheduleBase & {
+    intervalType: 'DAILY';
+};
+
+export type WeeklySchedule = ScheduleBase & {
+    intervalType: 'WEEKLY';
+    daysOfWeek: { days: number[] };
+};
+
+export type MonthlySchedule = ScheduleBase & {
+    intervalType: 'MONTHLY';
+    daysOfMonth: { days: number[] };
+};
+
+export type Schedule = UnsetSchedule | DailySchedule | WeeklySchedule | MonthlySchedule;
+
+export type IntervalType = Schedule['intervalType'];
+
+type BaseComplianceScanConfigurationSettings = {
+    oneTimeScan: boolean;
+    profiles: string[];
+    scanSchedule: Schedule;
+    description?: string;
+};
+
+export type ClusterScanStatus = {
+    clusterId: string;
+    errors: string[];
+    clusterName: string;
+};
+
+export type ComplianceScanConfiguration = {
+    id?: string;
```

Comment:
Is `id` optional for scenario of payload to POST request?

If yes, other forms and wizards initialize `id: ''` in the initial state so property can be required and prevent need for conditional rendering in uses like `key={id}` props for table rows.

Reply:
Correct, the `id` is optional for POST but required for PUT. So I actually like what you're suggesting. This is also more evidence for splitting up the POST and PUT service methods.